### PR TITLE
Unify L4 IPv6 shared health check firewall source ranges

### PR DESCRIPTION
**Pull request by FelipeYepez**
Unify the internal (ILB) and external (NetLB) GCE health check IPv6 source ranges in shared ETP Cluster firewall rules. This resolves the active race condition where the controllers constantly overwrite each other's distinct ranges in GCE. https://docs.cloud.google.com/load-balancing/docs/health-check-concepts

Unique ETP Local firewall rules remain strictly isolated to their specific range for defense-in-depth security.

Added a dedicated unit test suite (TestEnsureHealthCheckWithIPv6OnlyFirewalls) to verify all combinations, including a live-matching reconciliation edge case.

### CODE REVIEW

File: pkg/l4/healthchecks/healthchecksl4_test.go [hunk 1]
```diff
@@ -766,6 +767,196 @@ func TestEnsureHealthCheckWithDualStackFirewalls(t *testing.T) {
...
+
+			hcs := NewL4HealthChecks(fakeGCE, &record.FakeRecorder{}, klog.TODO(), true)
+			if tc.existingHC != nil {
+				err := hcs.hcProvider.Create(tc.existingHC)
+				if err != nil {
+					t.Fatalf("hcProvider.Create() err=%v", err)
+				}
+			}
+			if tc.existingFirewall != nil {
+				fakeGCE.CreateFirewall(tc.existingFirewall)
```
Comment (TortillaZHawaii) [inline pkg/l4/healthchecks/healthchecksl4_test.go]: no err check?

File: pkg/l4/healthchecks/healthchecksl4_test.go [hunk 2]
```diff
@@ -766,6 +767,196 @@ func TestEnsureHealthCheckWithDualStackFirewalls(t *testing.T) {
...
+
+	testCases := []struct {
+		desc               string
+		existingHC         *composite.HealthCheck
+		existingFirewall   *compute.Firewall
+		svc                *corev1.Service
+		wantHC             *composite.HealthCheck
+		wantUpdate         utils.ResourceSyncStatus
+		wantUpdateFw       utils.ResourceSyncStatus
+		expectedIPv6Ranges []string
```
Comment (TortillaZHawaii) [inline pkg/l4/healthchecks/healthchecksl4_test.go]: why all of them are`want...` but this is `expected...`?

**Review by TortillaZHawaii — COMMENTED** [review]
looks good

File: pkg/l4/healthchecks/healthchecksl4_test.go (hunk 2 above)
Comment (FelipeYepez) [reply pkg/l4/healthchecks/healthchecksl4_test.go]: Renamed

File: pkg/l4/healthchecks/healthchecksl4_test.go (hunk 1 above)
Comment (FelipeYepez) [reply pkg/l4/healthchecks/healthchecksl4_test.go]: Added the check, thanks

File: pkg/l4/healthchecks/healthchecksl4_test.go [hunk 3]
```diff
@@ -766,6 +767,199 @@ func TestEnsureHealthCheckWithDualStackFirewalls(t *testing.T) {
...
+		{
+			desc:           "NetLB ETP Local IPv6-only (Isolated NetLB range)",
+			svc:            svcNetLBLocal,
+			wantHC:         newL4HealthCheck(l4Namer.L4HealthCheck(svcNetLBLocal.Namespace, svcNetLBLocal.Name, false), namespacedNameNetLBLocal, false, hcDefaultPath, 1234, utils.XLB, meta.Global, testClusterValues.Region, klog.TODO()),
+			wantUpdate:     utils.ResourceUpdate,
+			wantUpdateFw:   utils.ResourceUpdate,
+			wantIPv6Ranges: []string{L4NetLBIPv6HCRange},
+		},
+		{
+			desc:           "NetLB ETP Cluster IPv6-only (Merged ranges - Smart Unification)",
```
Comment (TortillaZHawaii) [inline pkg/l4/healthchecks/healthchecksl4_test.go:848]: sounds very ai

Comment (FelipeYepez) [reply pkg/l4/healthchecks/healthchecksl4_test.go:848]: Of course it is. Busted :)
